### DISCUSSION TITLE
manually setting the task instance label property

### DISCUSSION THREAD

**Discussion by HugoCornu** [apache/airflow#25478] [Q&A]
I have found that the .label property of an operator can be set.
This allow to have an arbitrary name in the graph DAG visualisation. 
In particular this allow two or more tasks to have the same name which is clearer than unique names when both tasks do the same thing.

It this OK ? Or could it lead to some issues ?

**Comment by potiuk** [comment]
I do not understand what you are asking for. Example would be nice.

**Reply by HugoCornu** [reply]
Hi Potiuk

Here is an example showing how to set a python property : 
```
t0 = SomeOperator(task_id='must_be_unique_0')
t0.label = "can_be_arbitrary" 
t1 = SomeOperator(task_id='must_be_unique_1')
t1.label = "can_be_arbitrary"
```

I know it works but it does not seems that this usage was intended. Normally the label property is set by the TaskMixin class. The label property is not described in airflow docs (or I have not found where)
Is this OK ? or could it lead to problems ? Is it safe ? Will it break something that I am not aware of ?
It's a technical question. Let me know if it is still not clear.

Is there a better way or proper way to achieve the same result ? (DAG visualisation with some tasks sharing the same name)

**Reply by potiuk** [reply]
Like everything in Python you can modify any objects. Which does not mean you should.

Label is one of the automatically calculated properties. It is not documented what happens if you change it. Don't do it. It can lead to problems. It will break things you are not aware of. And even things I am not aware of, so simply don't do it. 

No there is no way to do it differently now. 

There are opened discussions that were discussing similar features (you can look it up searching in issues). But  - as everything in open-source - if there are only discussions but no real work behind it, nothing gets done. 

So if you think you need that feature, I am encouraging you to take a closer look and implement it in a form of PR proposal - you could become one of the > 2100 contributors - mostly people likel you who had a need and contributed their proposal back. 

But just to give you a glimpse of what it is - it's not as easy as dynamically changing one property. We rely onthe property likely in many places where we display information about the task and tasks instance have many variations and contexts. 

You will have to make sure things will work properly for a  number of cases - we have subdags, we have task gtroups we have dynamically mapped tasks) - in all of them there are some ways task information is displayed and adding custom name there has to be carefully implemented to include this. Additionally probably API has to be updated to include that capability and some way of handling task deferrals and serializing this information to DB would have to be added as well.

It does not seem complex - but again - if no-one does it, it does not get done, so if  you really think you need, I encourage you to contribute (or maybe you can find and convince someone else to do it if you don't feel like it).

**Reply by HugoCornu** [reply]
thanks for your answer
